firedrake/slate/slac/kernel_builder.py

```diff
@@ -32,6 +41,8 @@
 :param local_temp: The local temporary for the coefficient vector.
 """
 
+SCALAR_TYPE = "double"
```

Comment:
This is probably gonna hurt when doing complex numbers.

Reply:
Yes, there are few other places where I didn't do things complex safe.